Services/data/devicemodels/scripts/prototype-02-state.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+/*global log*/
+/*global updateState*/
+/*global updateProperty*/
+/*global sleep*/
+/*jslint node: true*/
+
+"use strict";
+
+var center_latitude = 47.620433;
+var center_longitude = -122.350987;
+
+// Default state
+var state = {
+    online: true,
+    temperature: 85.0,
+    temperature_unit: "F",
+    pressure: 150.0,
+    pressure_unit: "psig",
+    latitude: center_latitude,
+    longitude: center_longitude
+};
+
+// Default device properties
+var properties = {};
+
+/**
+ * Restore the global state using data from the previous iteration.
+ *
+ * @param previousState device state from the previous iteration
+ * @param previousProperties device properties from the previous iteration
+ */
+function restoreSimulation(previousState, previousProperties) {
+    // If the previous state is null, force a default state
+    if (previousState) {
+        state = previousState;
+    } else {
+        log("Using default state");
+    }
+
+    if (previousProperties) {
+        properties = previousProperties;
+    } else {
+        log("Using default properties");
+    }
+}
+
+/**
+ * Simple formula generating a random value around the average
+ * in between min and max
+ *
+ * @returns random value with given parameters
+ */
+function vary(avg, percentage, min, max) {
```

Comment:
this function is repeated in all files, can it be moved to some common util?

Reply:
Not 100% sure how this would play out with simulation. Added a bug https://msazure.visualstudio.com/One/_workitems/edit/2304578